File: scala/investigation/scala_for_impatient/src/com/zetsubou_0/scala/test/Person.scala
package com.zetsubou_0.scala.test

/**
 * Created by Kiryl_Lutsyk on 10/26/2015.
 */

class Person(private[this] val name: String) {
    private val MIN_AGE = 18
    private var privateAge = 0
    private val counter = new Counter(0)
    private[this] val names: Array[String] = name.split(" ")

    def firstName = names(0)

    def lastName = names(1)

    def age = privateAge

    def age_=(newValue: Int) {
        if (newValue > MIN_AGE) privateAge = newValue
    }

    def action(person: Person): Unit = {
        person.counter.increment()
        Console.println(person.counter.current)
    }
}

File: java/investigation/src/main/java/com/zetsubou_0/testdemo/bean/Person.java
package com.zetsubou_0.testdemo.bean;

/**
 * Created by Kiryl_Lutsyk on 12/2/2015.
 */
public class Person {
    private final String firstName;
    private final String lastName;
    private int age;

    public Person(final String firstName, final String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(final int age) {
        this.age = age;
    }
}
